src/workerd/api/js-rpc-service.h

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2017-2023 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+#pragma once
+
+#include <workerd/api/js-rpc-target.h>
+
+namespace workerd::api {
+
+class JsRpcServiceEvent final: public ExtendableEvent {
+public:
+  explicit JsRpcServiceEvent();
+
+  static jsg::Ref<JsRpcServiceEvent> constructor(kj::String type) = delete;
```

Comment:
```suggestion
  static jsg::Ref<JsRpcServiceEvent> constructor() = delete;
```

Reply:
The purpose here is to block visibility of `ExtendableEvent::constructor()`, which itself does take a string parameter, though matching the parameter list isn't actually necessary. It looks like other events that extend `ExtendableEvent` but need to block the constructor use the same phrasing as here.

Reply:
Yep. I forgot to clarify that this comment is non-blocking.

It turns out that listing the parameters isn't necessary to actually hide it. To reduce some boilerplate around this pattern, I've been considering introducing a new macro:

```
#define JSG_DISALLOW_NEW(Name) static jsg::Ref<Name> constructor() = delete
```

Allowing for...

```
class FooEvent : public Event {
public:
  JSG_DISALLOW_NEW(FooEvent);
  // ...
};
```